test/ContainerTest.php

```diff
@@ -217,15 +219,40 @@ public function testMultipleCallsToExpirationSecondsAggregates()
         $this->container->bar = 'baz';
         $this->container->baz = 'bat';
         $this->container->bat = 'bas';
+        $currentTimestamp     = time();
         $this->container->setExpirationSeconds(3600);
         $this->container->setExpirationSeconds(1800, 'foo');
         $this->container->setExpirationSeconds(900, ['baz', 'bat']);
         $storage = $this->manager->getStorage();
         $metadata = $storage->getMetadata($this->container->getName());
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 1800, $metadata['EXPIRE_KEYS']['foo']);
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 900, $metadata['EXPIRE_KEYS']['baz']);
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 900, $metadata['EXPIRE_KEYS']['bat']);
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 3600, $metadata['EXPIRE']);
+        $this->assertEquals($currentTimestamp + 1800, $metadata['EXPIRE_KEYS']['foo']);
+        $this->assertEquals($currentTimestamp + 900, $metadata['EXPIRE_KEYS']['baz']);
+        $this->assertEquals($currentTimestamp + 900, $metadata['EXPIRE_KEYS']['bat']);
+        $this->assertEquals($currentTimestamp + 3600, $metadata['EXPIRE']);
+    }
+
+    public function testSettingExpirationSecondsUsesCurrentTime()
+    {
+        sleep(3);
+        $this->container->setExpirationSeconds(2);
+        $this->container->foo = 'bar';
+
+        // Simulate a second request, backup the request time and overwrite it with current time()
+        $requestTimeBackup      = $_SERVER['REQUEST_TIME'];
+        $requestTimeFloatBackup = $_SERVER['REQUEST_TIME_FLOAT'] ?? null;
+
+        $_SERVER['REQUEST_TIME']       = time();
+        $_SERVER['REQUEST_TIME_FLOAT'] = microtime(true);
+
+        try {
+            $this->assertEquals('bar', $this->container->foo);
+        } catch (\Throwable $e) {
+            // Restore the original values
+            $_SERVER['REQUEST_TIME']       = $requestTimeBackup;
+            $_SERVER['REQUEST_TIME_FLOAT'] = $requestTimeFloatBackup;
```

Comment:
I don't think we need to do this; PHPUnit restores superglobals between test executions by default, IIRC.

I'll try removing it locally when I merge.